### PR TITLE
add support for ActiveRecord 4.2

### PR DESCRIPTION
**Pull request by hasghari**

### CODE REVIEW

File: lib/arel-helpers/join_association.rb [hunk 1]
```diff
@@ -62,6 +64,22 @@ def join_association_4_1(table, association, join_type)
           join_type.new(constraint.left, right)
         end
       end
+
+      def join_association_4_2(table, association, join_type)
+        associations = association.is_a?(Array) ? association : [association]
+        join_dependency = ActiveRecord::Associations::JoinDependency.new(table, associations, [])
+
+        join_dependency.join_constraints([]).map do |constraint|
+          join = constraint.joins[0]
```
Comment (bastjan) [inline lib/arel-helpers/join_association.rb]: in some cases constraint.joins is an array with multiple joins. for example: `thing has and belongs to many things`

I believe something like
```ruby
constraints.map do
  joins.map do
    join_type.new
  end
end.flatten
```
should solve the issue.

**Comment by hasghari** [comment]
@bastjan Good catch. I've updated the pull request with your suggested change.

**Comment by camertron** [comment]
Oops, sorry for taking so long to respond. Looks good to me :) :+1:

**Comment by camertron** [comment]
v2.1.0 published

**Pull request merged**